packages/controls/src/controls/checkbox/checkbox.ts

```diff
@@ -152,6 +157,16 @@ class Definition<C extends Config> extends ControlDefinition<
     return data
   }
 
+  resolveValue(
+    data: DataType<C> | undefined,
+  ): Resolvable<ResolvedValueType<C> | undefined> {
+    return {
+      readStableValue: (_previous?: ResolvedValueType<C>) =>
+        this.fromData(data) ?? this.config.defaultValue,
+      subscribe: () => () => {},
```

Comment:
`subscribe` is meant to notify the subscribers if the value changes "on its own", without a change to the original data -- for example when a color's swatch gets edited. None of the "basic" controls in this PR have external resource dependencies that could trigger this situation, thus a no-op implementation.

Reply:
That seems like a weird additional way for things to change that seems unnecessary, though, would you agree? Wouldn't a subscriber just want to know "did this value change"?

Reply:
Well, yes, what I am saying the only way the value can change if it depends on something external, which is not the case for any of the basic controls.

Reply:
I guess it all depends on how long lives this object is. Seems that whenever the data changes we create a new one. For something like a color where the resolved value can change even when the data doesn't change I can see why the subscription is necessary. I think that in the future it would be nice to have the lifetime of the resolved object state be the same as the data in the store, so subscribing to this value is not just for external changes, but for data changes too since we wouldn't be constructing a new observable/subscribable instance when the data changes. In fact there would only be one subscribe, for the entire store's state

Reply:
Right, that is the design direction where we put everything in the Redux store (somehow). This and the upcoming follow-up PRs are implementing what I thought we've agreed upon as an intermediate step towards that, in which the state is managed in React.